### PR TITLE
chore(actions): update and restructure actions

### PR DESCRIPTION
**Pull request by supermaxiste**
Changes:

- All actions are updated to their latest version 
- All `yml` files are using consistent formatting and versions
- `docker-publish` now triggers only for build&push for any changes to `main` (before it would build+build&push)
- `sphinx-docs` now triggers build for PRs when `docs/**` changes and build+push for any changes to `main`

### CODE REVIEW

File: .github/workflows/sphinx-docs.yml [hunk 1]
```diff
@@ -1,14 +1,47 @@
 name: docs
-on: [push, pull_request, workflow_dispatch]
+on:
+  push:
+    branches: [main]
```
Comment (cmdoret) [inline .github/workflows/sphinx-docs.yml:4]: Ah great! Thanks for this :)

**Review by cmdoret — APPROVED** [review]
Awesome, looks much better now! Thanks for taking the time to bump action versions :raised_hands:

**Comment by rmfranken** [comment]
Nice work guys!! Kinda curious what's going on with this, seems like an awefully long time to publish no?
image

soon we can implement the new and released gimie on IP 😄

**Comment by supermaxiste** [comment]
@rmfranken it is indeed long, but this action will not trigger most of the time 😄 currently it's setup to trigger only when dependencies, github action files or dockerfiles change. 
Note that the title of the action has "Publish" in it, but in this case it's the building the container without publishing. Publishing happens only for changes to main.

**Pull request merged**